javalib/src/main/scala/java/lang/Character.scala

```diff
@@ -1662,429 +1963,467 @@ object Character {
+      "adlam"                                      -> ADLAM,
+      "indicsiyaqnumbers"                          -> INDIC_SIYAQ_NUMBERS,
+      "ottomansiyaqnumbers"                        -> OTTOMAN_SIYAQ_NUMBERS,
+      "arabicmathematicalalphabeticsymbols"        -> ARABIC_MATHEMATICAL_ALPHABETIC_SYMBOLS,
+      "mahjongtiles"                               -> MAHJONG_TILES,
+      "dominotiles"                                -> DOMINO_TILES,
+      "playingcards"                               -> PLAYING_CARDS,
+      "enclosedalphanumericsupplement"             -> ENCLOSED_ALPHANUMERIC_SUPPLEMENT,
+      "enclosedideographicsupplement"              -> ENCLOSED_IDEOGRAPHIC_SUPPLEMENT,
+      "emoticons"                                  -> EMOTICONS,
+      "ornamentaldingbats"                         -> ORNAMENTAL_DINGBATS,
+      "alchemicalsymbols"                          -> ANATOLIAN_HIEROGLYPHS,
+      "geometricshapesextended"                    -> GEOMETRIC_SHAPES_EXTENDED,
+      "supplementalarrowsc"                        -> SUPPLEMENTAL_ARROWS_C,
+      "chesssymbols"                               -> CHESS_SYMBOLS,
+      "cjkunifiedideographsextensionb"             -> CJK_UNIFIED_IDEOGRAPHS_EXTENSION_B,
+      "cjkunifiedideographsextensionc"             -> CJK_UNIFIED_IDEOGRAPHS_EXTENSION_C,
+      "cjkunifiedideographsextensiond"             -> CJK_UNIFIED_IDEOGRAPHS_EXTENSION_D,
+      "cjkunifiedideographsextensione"             -> CJK_UNIFIED_IDEOGRAPHS_EXTENSION_E,
+      "cjkunifiedideographsextensionf"             -> CJK_UNIFIED_IDEOGRAPHS_EXTENSION_F,
+      "cjkcompatibilityideographssupplement"       -> CJK_COMPATIBILITY_IDEOGRAPHS_SUPPLEMENT,
+      "tags"                                       -> TAGS,
+      "variationselectorssupplement"               -> VARIATION_SELECTORS_SUPPLEMENT,
+      "supplementaryprivateuseareaa"               -> SUPPLEMENTARY_PRIVATE_USE_AREA_A,
+      "supplementaryprivateuseareab"               -> SUPPLEMENTARY_PRIVATE_USE_AREA_B
+    )
+
+    // From: https://unicode.org/Public/UCD/latest/ucd/Blocks.txt
+    // Note:   When comparing block names, casing, whitespace, hyphens,
+    //         and underbars are ignored.
+    //         For example, "Latin Extended-A" and "latin extended a" are equivalent.
+    //         For more information on the comparison of property values,
+    //           see UAX #44: http://www.unicode.org/reports/tr44/
+    private def toNormalizedName(name: String): String = {
+      name.toLowerCase.replaceAll(raw"[\s\-_]", "")
+    }
+
+    ////////////////
+    // End Generated
+    ////////////////
 
     def forName(blockName: String): UnicodeBlock = {
-      if (blockName == null) {
-        throw new NullPointerException()
-      }
-      BLOCKS_BY_NAME
-        .get(blockName)
-        .fold {
-          throw new IllegalArgumentException()
-        } { value => value }
+      blocksByNormalizedName
```

Comment:
Why remove a perfectly good & useful null check?
There is the off chance is that such was recommended by someone in the know at
Scala.js.

If not, my long training in expecting explicit argument checks for every & anything passed in from application code leads me to miss the null check here.  True, that
some code some-J-Random-where, perhaps 5 layers down will throw the exception
at some hard to find and debug place.  With line numbers in the NIR for SN in the offing, there is a chance to get an explicit line number at an understandable place early in execution, when there is less shrapnel.

Do I sound opinionated here?  I was merely trying for experienced.  I am also aware
that I am a contributor, and that more senior reviewers set the standard & practice.